index.rst

```diff
@@ -96,11 +102,173 @@ The stated requirement is: Must support a system so that data may be referred to
 
-Configuration
+Butler
+------
+The ``Butler`` class is the  overall interface and manager for repositories. 
+The Butler init function takes a list of input and output repositories (see
+below for a description of inputs and outputs) that are used as locations for
+i/o.
+
+The new butler API is ``Butler(inputs=None, outputs=None)``. Values for both
+``inputs`` and ``outputs`` can be an instance of the ``RepositoryArgs`` class or
+can be a string. Additionally, the value can be either a single item or many
+items in a sequence container. If the value is only a string, it is treated as a
+URI. In inputs it must refer to a persisted ``RepositoryCfg`` and in outputs it
+can refer to an existing ``RepositoryCfg`` or can be a location to create a new
+repository.
+
+Inputs and Outputs
+^^^^^^^^^^^^^^^^^^
+
+Butler will only perform read actions on input repositories and will perform
+read and write actions on output repositories. Repositories may also have a
+mode that can be one of:
+
+* read
+* write
+* read-write 
+
+When the mode of an output repository is read-write it will also be used as an
+input. Attempting to pass a read-only repo as a butler output or a write-only
+repo as a butler input will raise an exception.
+
+
+Already existing Repositories as Outputs
+""""""""""""""""""""""""""""""""""""""""
+
+When initializing a ``Butler`` with one or more output repositories that already
+exist it is important to not add any input repositories that are not already
+parents of the output repository. This is because all butler inputs (including
+readable outputs) become parents of output repositories, and the repository
+configuration is not allowed to change after it has been written.
+
+Output arguments derived from inputs
+""""""""""""""""""""""""""""""""""""
+
+Some settings for output repositories can be derived from input repository
+configurations. For example, if an output configuration does not specify a
+mapper, the input mapper may possibly be assumed (this will work as long as all
+the input repositories use the same type of mapper; if the inputs use different
+types of mapper then a single type mapper can not be inferred to use for the
+output repositories). When possible the butler will use settings from input
+configurations to complete RepositoryArgs parameters for output repositories. 
+
+Search Order
+""""""""""""
+The order of repositories passed to inputs and outputs is meaningful; search is
+depth-first and in order (left to right). Readable outputs will be searched
+before inputs. Parents of readable outputs/inputs will be searched before the
+next passed-in output/input.
+
+Tagging 
+^^^^^^^
+
+Readable repositories can be “tagged” with a non-persisted id that gets used
+when reading from a repository. A tag can be specified in ``RepositoryArgs``
+when initializing a Butler. A repository can be tagged with more than one tag by
+passing in a container of tags. The tag is not persisted with the repository.
```

Comment:
Since you say "not persisted" here, maybe no need to say it earlier in this paragraph?